ClientApp/viking-vault/src/Router.tsx

```diff
@@ -0,0 +1,66 @@
+import React from "react";
+import { Route, Redirect, BrowserRouter as BRouter } from 'react-router-dom';
+import { RegisterForm } from "./RegisterForm";
+import { LoginForm } from "./Login";
+import UserPage from './components/UserPage';
+import {constants} from "./ConstantVariables";
+import "./App.css"
+
+const baseUrl = constants.baseUrl;
+
+interface IState {
+    isAdmin: boolean;
+}
+
+class Router extends React.Component<any, IState> {
+    state = {
+        isAdmin: false
+    }
+
+    private isUserAdmin() {
+        let token = sessionStorage.getItem("Authentication-Token");
+        if(token === null) {
+            return;
+        }
+        fetch(baseUrl+"admin", {
+            method: "GET",
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+              'x-access-token' : token.toString()
+            }
+        }).then(response => response.json())
+        .then(result => {
+            if(result === true) {
+                this.setState({
```

Comment:
Please also get rid of the if/else statements in here and replace them with `this.setState({ isAdmin: result })`